packages/xo-server-netbox/src/index.js

```diff
@@ -284,22 +291,27 @@ class Netbox {
       }
     }
 
-    // FIXME: Should we deduplicate cluster names even though it also fails
-    //        when a cluster within another cluster type has the same name?
     const newClusters = []
-    if (clustersToUpdate.length > 0) {
-      log.info(`Updating ${clustersToUpdate.length} clusters`)
-      newClusters.push(...(await this.#request('/virtualization/clusters/', 'PATCH', clustersToUpdate)))
-    }
-    if (clustersToCreate.length > 0) {
-      log.info(`Creating ${clustersToCreate.length} clusters`)
-      newClusters.push(...(await this.#request('/virtualization/clusters/', 'POST', clustersToCreate)))
-    }
+    newClusters.push(...(await this.#request('/virtualization/clusters/', 'PATCH', clustersToUpdate)))
+    newClusters.push(...(await this.#request('/virtualization/clusters/', 'POST', clustersToCreate)))
+
```

Comment:
Commit: [c314090](https://github.com/vatesfr/xen-orchestra/pull/6989/commits/c314090fcdfd2d4a022dcc916852b59eb96e7320)
Is it necessary to keep the order of these requests, or can we group them together with a `Promise.all`?
```suggestion
    ;(
      await Promise.all([
        this.#request('/virtualization/clusters/', 'PATCH', clustersToUpdate),
        this.#request('/virtualization/clusters/', 'POST', clustersToCreate),
      ])
    ).forEach(result => newClusters.push(...result))
```

Reply:
The order is important. For example, if you have:

Sync 1:
```
{ id: 1, name: 'Foo' }
```
Sync 2:
```
{ id: 1, name: 'Bar' }
{ id: 2, name: 'Foo' }
```
Then during Sync 2, you need to make sure object `1` is updated before object `2` is created, otherwise you'd get a name conflict.

So the order is always `(DELETE >) PATCH > POST`.